src/main/starlark/builtins_bzl/common/python/py_executable_bazel.bzl

```diff
@@ -290,11 +290,13 @@ def _expand_bootstrap_template(
 
     if runtime:
         shebang = runtime.stub_shebang
+        template = runtime.bootstrap_template
```

Comment:
It looks like just one test is failing because bootstrap template is None, which makes sense -- the test is specifically testing a manually created PyRuntimeInfo. It's expected that PyRuntimeInfo.bootstrap_template is set; Normally this is handled by the py_runtime rule, but the test is specifically checking for a custom implementation.

This is around the line you'll need to modify:
https://cs.opensource.google/bazel/bazel/+/master:src/test/java/com/google/devtools/build/lib/rules/python/PythonStarlarkApiTest.java;l=120;drc=17827eee79808fceb6558d632c5b5594e8e9a47a

The net changes should be something like:

* In the call to PyRuntimeInfo(), add `bootstrap_template=ctx.file.bootstrap_template`
* A couple lines below, to the `userruntime` rule definition, add `'bootstrap_template': attr.label_list(allow_files=True)`
* A couple lines below, in the call to `userruntime()` add `bootstrap_template='bootstrap.txt'`

And `bazel test //src/test/java/com/google/devtools/build/lib/rules/python:PythonStarlarkApiTest `